app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt

```diff
@@ -0,0 +1,184 @@
+package org.fossasia.susi.ai.login
+
+import android.app.ProgressDialog
+import android.content.Intent
+import android.graphics.Color
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_login.*
+import org.fossasia.susi.ai.R
+import org.fossasia.susi.ai.activities.ForgotPasswordActivity
+import org.fossasia.susi.ai.activities.MainActivity
+import org.fossasia.susi.ai.activities.SignUpActivity
+import org.fossasia.susi.ai.helper.AlertboxHelper
+import org.fossasia.susi.ai.helper.Constant
+
+/**
+ * <h1>The Login activity.</h1>
+ * <h2>This activity is used to login into the app.</h2>
+ *
+ * Created by chiragw15 on 4/7/17.
+ */
+class LoginActivity : AppCompatActivity(), ILoginView {
+
+    var loginPresenter: ILoginPresenter? = null
+    var progressDialog: ProgressDialog? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        addListeners()
+
+        if (savedInstanceState != null) {
+            email.editText!!.setText(savedInstanceState.getCharSequenceArray(Constant.SAVED_STATES)!![0].toString())
```

Comment:
There are 4 reference of "!!" operator including this one. I have changed two of them and will include them in next PR. The other two are related to Preference Manager. Will change it when we shift it.